modules/lfs/server.go

```diff
@@ -415,7 +413,7 @@ func Represent(rv *RequestVars, meta *models.LFSMetaObject, download, upload boo
 func ContentMatcher(r macaron.Request) bool {
 	mediaParts := strings.Split(r.Header.Get("Accept"), ";")
 	mt := mediaParts[0]
-	return mt == contentMediaType
+	return mt != metaMediaType
```

Comment:
why does it checks not equal now? imho it should still be:
```suggestion
	return mt == metaMediaType
```

Reply:
It was previously checking if mt == "application/vnd.git-lfs"

now it checks if mt != "application/vnd.git-lfs+json"

Reply:
Is this really correct then as any invalid type would also match

Reply:
I'm not sure that the spec says much at all about this - what it doesn't do is specify that it must be the current content type. @slonopotamus any thoughts?

Reply:
I believe accepting any content type for blob is OK. It allows downloading blobs with just a browser. You already have "filename" parameter that skips content-type check. What worries me is `verify` url. Does it still have ContentMatcher? If yes, this goes against the spec

Reply:
https://github.com/zeripath/gitea/blob/ab05d5d881d60fe5342af8607e7ef14b0e6fa7ed/modules/lfs/server.go#L103

You don't need `ContentMatcher` here.

I actually wonder if you need `ContentMatcher` *at all*.

Reply:
https://github.com/zeripath/gitea/blob/ab05d5d881d60fe5342af8607e7ef14b0e6fa7ed/modules/lfs/server.go#L350

So verify still has the content matcher - are you suggesting that the content type for verify needs to be the meta content type?

Reply:
https://github.com/git-lfs/git-lfs/blob/master/docs/api/basic-transfers.md#downloads

Relevant part of spec. Object upload/download doesn't insist on any `Accept` headers. Only object verification has "application/vnd.git-lfs+json". So yes, I believe that `MetaMatcher` is correct thing for `verify`.

As I understand the spec, the (not stated anywhere) rule is: `Accept: application/vnd.git-lfs+json` is required if and only if response format is going to be JSON.

I also believe that current (before this PR) LFS implementation goes against HTTP conventions. Suppose that client sends you `Accept: application/vnd.git-lfs`. But you are **not** responding with `Content-Type: application/vnd.git-lfs`! You're responding with `Content-Type: application/vnd.git-lfs+json` for `verify` and `Content-Type: application/octet-stream` for blob downloads. Does it make any sense to install Accept header with type that you never respond with? My answer is no.

Reply:
So the verify problem is interesting - how is a verification request ever accepted by Gitea? Either they're all failing or git-lfs must have been listening to https://github.com/go-gitea/gitea/blob/6eb53ac570ab9af51fc9cbd79f1db782edce57e0/modules/lfs/server.go#L388.

I'm guessing that the header here https://github.com/zeripath/gitea/blob/ab05d5d881d60fe5342af8607e7ef14b0e6fa7ed/modules/lfs/server.go#L405 needs to include an accept meta type and that 350 needs to be metaMatcher.

Reply:
You want to say that you pass tests without installing any `Accept` header for `verify` and having `ContentMatcher` in it? Oh my. I start to suspect that git-lfs does *not* install `Accept` header for `verify` unless told by server to do so.

We may possibly want to wait for reaction from LFS devs to git-lfs/lfs-test-server#85.

Reply:
It's ok I've just pushed changes to the branch as above.

Now it's probably the case that our tests are insufficient. For example in #6999 and #6961 I noticed that SSH locks weren't working. I don't think we attempt an LFS checkout either.

Reply:
Hmm interestingly that breaks the media endpoint...

Reply:
Reported git-lfs/git-lfs#3662 to git-lfs so they start to send `Accept: application/vnd.git-lfs+json` for `/verify` URL.

Reply:
I suddenly understood that it is only me using "/verify URL" term. This is because in git-as-svn we use *completely different* URLS for verification and upload/download, thus avoiding the need to determine what to do based on `Accept` HTTP header.